### PR TITLE
feat(common & core & legacy & python/stellar): add support for `ClaimClaimableBalanceOp`.

### LINKED ISSUE #3433: Stellar: add support for `ClaimClaimableBalanceOp`.
`ClaimClaimableBalanceOp` allows users to claim pending balances. For more detailed instructions, please refer to the Spec section.

**Spec:**

[Claimable Balances](https://developers.stellar.org/docs/encyclopedia/claimable-balances)
[Stellar CAP-0023](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0023.md)

XDR Spec:
```c
struct ClaimClaimableBalanceOp
{
    ClaimableBalanceID balanceID;
};

enum ClaimableBalanceIDType
{
    CLAIMABLE_BALANCE_ID_TYPE_V0 = 0
};

union ClaimableBalanceID switch (ClaimableBalanceIDType type)
{
case CLAIMABLE_BALANCE_ID_TYPE_V0:
    Hash v0;
};
```

**Design:**

 TT

T1

### PR DESCRIPTION
**Pull request by overcat**
Closes #3433 

It has been two years since I last submitted code. I hope I haven't missed anything. If there is, please let me know. Thank you 🙏

### CODE REVIEW

**Review by prusnak — DISMISSED** [review]

File: python/tests/test_stellar.py [hunk 1]
```diff
@@ -924,3 +925,21 @@ def test_tx_source_muxed_account_not_support_raise():
...
+
+    envelope = tx.append_claim_claimable_balance_op(
+        balance_id == balance_id, source=operation_source
+    ).build()
+
+    tx, operations = stellar.from_envelope(envelope)
+    assert len(operations) == 1
+    assert isinstance(operations[0], messages.StellarClaimClaimableBalanceOp)
+    assert operations[0].source_account == operation_source
+    assert operations[0].balance_id == binascii.unhexlify(balance_id)
```
[the remark this answers is not in the document]
Comment (overcat) [reply python/tests/test_stellar.py]: 161324ade0bd91185dbdd4d1273763dc421f4edd

File: python/src/trezorlib/stellar.py [hunk 2]
```diff
@@ -272,6 +275,11 @@ def _read_operation(op: "Operation") -> "StellarMessageType":
             destination_min=_read_amount(op.dest_min),
             paths=[_read_asset(asset) for asset in op.path],
         )
+    if isinstance(op, ClaimClaimableBalance):
+        return messages.StellarClaimClaimableBalanceOp(
+            source_account=source_account,
+            balance_id=binascii.unhexlify(op.balance_id),
```
[the remark this answers is not in the document]
Comment (overcat) [reply python/src/trezorlib/stellar.py]: 161324ade0bd91185dbdd4d1273763dc421f4edd

File: core/src/apps/stellar/operations/layout.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import binascii
```
[the remark this answers is not in the document]
Comment (overcat) [reply core/src/apps/stellar/operations/layout.py]: 61d7f3ea96b386bb5f5b3839958e78642b769fc7

File: common/protob/messages.proto [hunk 4]
```diff
@@ -256,6 +256,8 @@ enum MessageType {
     MessageType_StellarBumpSequenceOp = 221 [(wire_in) = true];
     MessageType_StellarManageBuyOfferOp = 222 [(wire_in) = true];
     MessageType_StellarPathPaymentStrictSendOp = 223 [(wire_in) = true];
+    // omitted: StellarCreateClaimableBalanceOp is not a supported operation, would be 224
```
Comment (matejcik) [inline common/protob/messages.proto:259]:
```suggestion
    reserved 224;  // omitted: StellarCreateClaimableBalanceOp
```
(please make the same change for InflationOp above while you're at it)

**Review by matejcik — COMMENTED** [review]
please also regenerate the UI fixtures for model R (`export TREZOR_MODEL=R` before building core)

i'll need to confirm whether we'll support further Stellar changes to the legacy (trezor one) codebase

File: common/protob/messages.proto (hunk 4 above)
Comment (overcat) [reply common/protob/messages.proto]: d702b78afe1d5e6bf470c224102f9d431c742ad0

I also added the test for T1. We can decide whether to keep it after confirmation.

**Comment by overcat** [comment]
> > i'll need to confirm whether we'll support further Stellar changes to the legacy (trezor one) codebase
> 
> for now please remove the change from legacy. you will need to add the message name to `legacy/firmware/protob/Makefile` for exclusion.
> 
> please squash your fixups and rebase to get rid of the conflict

Done, PTAL.

**Comment by overcat** [comment]
It looks like we have passed the necessary tests, so I performed a rebase.

**Comment by matejcik** [comment]
> Hi @matejcik, we also need to skip the relevant tests on t1. I have added it in [42f9d72](https://github.com/trezor/trezor-firmware/commit/42f9d723a209e19e3a40ef23980efff87630c00a). Please let me know if there is a better way.

seems reasonable, thank you.

one last CI pass for the final version of the commit

**Review by matejcik — APPROVED** [review]

**Comment by matejcik** [comment]
and we're in. thanks for your work :100:

**Comment by overcat** [comment]
> and we're in. thanks for your work 💯

@matejcik Thank you for your support. Additionally, I would like to ask if the reason we no longer add new features to t1 is due to limitations in t1's performance? For example, insufficient Flash space?

**Comment by matejcik** [comment]
It's more of a question of maintenance load. With the Safe 3 out (same codebase as the TT), we are shifting T1 into maintenance mode, meaning we will keep it working but not add new features. This one is very small, sure, but we need to draw the line somewhere.

**Pull request merged**